### PR TITLE
Fix/73 refactor prometheus metrics

### PR DESCRIPTION
**Pull request by dmercuriali**
Replaced bookkeeper metric provider with Prometheus java client. Bookkeeper StatsLogger is still used for HerdDB metrics.

Refactored metrics:
backends_pending_requests   GAUGE
backends_stuck_requests_total   COUNTER
backends_connection_time_ns   SUMMARY [host (host_port), result(success/failure)], quantiles 0.5 0.9 0.99
backends_open_connections   GAUGE [host (host_port)]
backends_active_connections   GAUGE [host (host_port)]
backends_sent_requests_total   COUNTER [host (host_port)]

listeners_requests_total   COUNTER [listener (host_port)]
listeners_running_requests   GAUGE
listeners_user_requests_total   COUNTER [userId]

clients_current_connected   GAUGE

health_backend_status   GAUGE [host (host_port)] - 0/1

cache_hits_total   COUNTER
cache_misses_total   COUNTER
cache_payload_memory_usage_bytes   GAUGE [area (direct/heap)]
cache_total_memory_usage_bytes   GAUGE
cache_non_cachable_requests_total   COUNTER

### CODE REVIEW

File: carapace-server/src/main/java/org/carapaceproxy/server/ClientConnectionHandler.java [hunk 1]
```diff
@@ -177,10 +177,14 @@ public String getListenerHost() {
...
         return listenerPort;
     }
-    
+
     public boolean isSecure() {
         return secure;
     }
+    
+    public long getTotalRequestsCount() {
+        return (long) this.totalRequests.get();
```
Comment (eolivelli) [inline carapace-server/src/main/java/org/carapaceproxy/server/ClientConnectionHandler.java]: This is a problem.
It is better to return an int

File: carapace-server/src/main/java/org/carapaceproxy/server/backends/BackendHealthManager.java [hunk 2]
```diff
@@ -183,6 +164,12 @@ public void run() {
                 }
             }
             status.setLastProbe(checkResult);
+
+            if (status.isReportedAsUnreachable()) {
+                BACKEND_UPSTATUS_GAUGE.labels(bconf.getHost() + "_" + bconf.getHostPort()).set(0);
```
Comment (eolivelli) [inline carapace-server/src/main/java/org/carapaceproxy/server/backends/BackendHealthManager.java]: host + hostport?
Isn't it redundant?

Comment (dmercuriali) [reply carapace-server/src/main/java/org/carapaceproxy/server/backends/BackendHealthManager.java]: yes, fixed to bconf.getHost() + "_" + bconf.getPort()

**Comment by dmercuriali** [comment]
Had to fix all tests related to cache metrics: since metrics are now static they don't reset after each test. I added a new method to manually reset them (CacheStats#resetCacheMetrics)

**Pull request merged**